--- ./src/modules/packages/p.c	2026-07-22 23:06:38.399859937 +0530
+++ ./src/modules/packages/packages.c	2026-07-22 23:11:58.009041361 +0530
@@ -3,6 +3,7 @@
 #include "common/strutil.h"
 #include "detection/packages/packages.h"
 #include "modules/packages/packages.h"
+#include <dirent.h>
 
 bool ffPrintPackages(FFPackagesOptions* options) {
     FFPackagesResult counts = {};
@@ -10,12 +11,19 @@
 
     const char* error = ffDetectPackages(&counts, options);
 
+    struct dirent *entry;
+    uint32_t pkg_veiler_count = system("Veiler pkgs");
+
+    if ( pkg_veiler_count > 0 ) {
+        counts.veiler = pkg_veiler_count;
+    }
+
     if (error) {
         ffPrintError(FF_PACKAGES_MODULE_NAME, 0, &options->moduleArgs, FF_PRINT_TYPE_DEFAULT, "%s", error);
         return false;
     }
 
-    if (counts.all == 0) {
+    if (counts.all == 0 && counts.veiler == 0) {
         ffPrintError(FF_PACKAGES_MODULE_NAME, 0, &options->moduleArgs, FF_PRINT_TYPE_DEFAULT, "No packages from known package managers found");
         return false;
     }
@@ -114,6 +122,7 @@
             }
             ffStrbufAppendS(&output, ", ");
         };
+
         FF_PRINT_PACKAGE(pacstall)
         FF_PRINT_PACKAGE(paludis)
         FF_PRINT_PACKAGE(pisi)
@@ -135,6 +144,7 @@
         FF_PRINT_PACKAGE(sorcery)
         FF_PRINT_PACKAGE(winget)
         FF_PRINT_PACKAGE(xbps)
+        FF_PRINT_PACKAGE(veiler)
 
         assert(output.length >= 2); // counts.all > 0 guarantees that at least one package count was printed, which guarantees that ", " was appended at least once
         ffStrbufSubstrBefore(&output, output.length - 1);
