--- /usr/bin/synfetch	2026-06-08 21:52:19.520772758 +0530
+++ ./synfetch	2026-06-09 22:15:08.662261176 +0530
@@ -1121,6 +1121,9 @@
     elif command -v pacman >/dev/null; then
         count=$(pacman -Qq 2>/dev/null | wc -l)
         pkgs+=("$count (pacman)")
+    elif command -v Veiler >/dev/null; then
+        count=$(Veiler pkgs)
+        pkgs+=("$count (veiler)")
     elif command -v nix-store >/dev/null; then
         count=$(nix-store -q --requisites /run/current-system 2>/dev/null | wc -l || echo 0)
         pkgs+=("$count (nix)")
